components/elm/src/main/elmfates_interfaceMod.F90

```diff
@@ -796,17 +860,19 @@ subroutine init(this, bounds_proc )
                endif
             endif
 
+            if (flandusepftdat == '') then
             ! initialize static layers for reduced complexity FATES versions from HLM
-            this%fates(nc)%bc_in(s)%pft_areafrac(:)=0._r8
-            do m = surfpft_lb,surfpft_ub
-               ft = m-surfpft_lb
-               this%fates(nc)%bc_in(s)%pft_areafrac(ft)=wt_nat_patch(g,t,m)
-            end do
+               this%fates(nc)%bc_in(s)%pft_areafrac(:)=0._r8
```

Comment:
In my tests with CTSM, I'm running into an error because `this%fates(nc)%bc_in(s)%bc_in%pft_areafrac` isn't initialized—probably because the fixed biogeography parameter `use_fates_fixed_biogeog` is false. See [these lines](https://github.com/NGEET/fates/blob/c1a7453c8331277a156a0ee158a870c1b8598a9f/main/FatesInterfaceMod.F90#L567-L573) from NGEET/fates#1116.

@ckoven I think you might need to add an `allocate` call to handle cases where `hlm_use_fixed_biogeog` is false. It looks like this bit of the CTSM code is called regardless of whether we have fixed biogeography on.

Reply:
Hmm... I'm still getting the error when I set `use_fates_fixed_biogeog` to true. Investigating.

Reply:
It seemed like the problem might have been in `flandusepftdat`, so I tested different combinations of that and `use_fates_fixed_biogeog`.

Case A: `flandusepftdat` pointing to a file and `use_fates_fixed_biogeog` false: Works on Derecho, fails on Izumi (intel and nag, unhelpful error messages—not worrying about this for now).

Case B: `flandusepftdat = ''` and `use_fates_fixed_biogeog` false: Fails because of an unallocated array at `this%fates(nc)%bc_in(s)%pft_areafrac(:)=0._r8.

Case C: `flandusepftdat` pointing to a file and `use_fates_fixed_biogeog` true: On Izumi, fails as A. But whereas A worked on Derecho, C fails with lots of this kind of warning:
```
dec1322.hsn.de.hpc.ucar.edu 161:  Nan values for pftareafrac. dumping site info.
dec1322.hsn.de.hpc.ucar.edu 161:  ----------------------------------------
dec1322.hsn.de.hpc.ucar.edu 161:   Site Coordinates
dec1322.hsn.de.hpc.ucar.edu 161:  ----------------------------------------
dec1322.hsn.de.hpc.ucar.edu 161:  latitude                    =   -90.0000000000000
dec1322.hsn.de.hpc.ucar.edu 161:  longitude                   =    165.000000000000
```
and then this error:
```
dec1322.hsn.de.hpc.ucar.edu 161:  issue with patch area in EDinit  -10000.0000000000       0.000000000000000E+000
dec1322.hsn.de.hpc.ucar.edu 161:  state_vector   1.00000000000000       0.000000000000000E+000
dec1322.hsn.de.hpc.ucar.edu 161:   0.000000000000000E+000  0.000000000000000E+000  0.000000000000000E+000
dec1322.hsn.de.hpc.ucar.edu 161:  area_error  0.000000000000000E+000
dec1322.hsn.de.hpc.ucar.edu 161:  area_bareground   1.00000000000000
dec1322.hsn.de.hpc.ucar.edu 161:  sites(s)%area_pft(:,i_lu_state)           1  0.000000000000000E+000
dec1322.hsn.de.hpc.ucar.edu 161:   0.000000000000000E+000  0.000000000000000E+000  0.000000000000000E+000
dec1322.hsn.de.hpc.ucar.edu 161:   0.000000000000000E+000  0.000000000000000E+000  0.000000000000000E+000
dec1322.hsn.de.hpc.ucar.edu 161:   0.000000000000000E+000  0.000000000000000E+000  0.000000000000000E+000
dec1322.hsn.de.hpc.ucar.edu 161:   0.000000000000000E+000  0.000000000000000E+000
dec1322.hsn.de.hpc.ucar.edu 161:  sites(s)%area_pft(:,i_lu_state)           2  0.000000000000000E+000
dec1322.hsn.de.hpc.ucar.edu 161:   0.000000000000000E+000  0.000000000000000E+000  0.000000000000000E+000
dec1322.hsn.de.hpc.ucar.edu 161:   0.000000000000000E+000  0.000000000000000E+000  0.000000000000000E+000
dec1322.hsn.de.hpc.ucar.edu 161:   0.000000000000000E+000  0.000000000000000E+000  0.000000000000000E+000
dec1322.hsn.de.hpc.ucar.edu 161:   0.000000000000000E+000  0.000000000000000E+000
dec1322.hsn.de.hpc.ucar.edu 161:  sites(s)%area_pft(:,i_lu_state)           3  0.000000000000000E+000
dec1322.hsn.de.hpc.ucar.edu 161:   0.000000000000000E+000  0.000000000000000E+000  0.000000000000000E+000
dec1322.hsn.de.hpc.ucar.edu 161:   0.000000000000000E+000  0.000000000000000E+000  0.000000000000000E+000
dec1322.hsn.de.hpc.ucar.edu 161:   0.000000000000000E+000  0.000000000000000E+000  0.000000000000000E+000
dec1322.hsn.de.hpc.ucar.edu 161:   0.000000000000000E+000  0.000000000000000E+000
dec1322.hsn.de.hpc.ucar.edu 161:  sites(s)%area_pft(:,i_lu_state)           4  0.000000000000000E+000
dec1322.hsn.de.hpc.ucar.edu 161:   0.000000000000000E+000  0.000000000000000E+000  0.000000000000000E+000
dec1322.hsn.de.hpc.ucar.edu 161:   0.000000000000000E+000  0.000000000000000E+000  0.000000000000000E+000
dec1322.hsn.de.hpc.ucar.edu 161:   0.000000000000000E+000  0.000000000000000E+000  0.000000000000000E+000
dec1322.hsn.de.hpc.ucar.edu 161:   0.000000000000000E+000  0.000000000000000E+000
dec1322.hsn.de.hpc.ucar.edu 161:  sites(s)%area_pft(:,i_lu_state)           5  0.000000000000000E+000
dec1322.hsn.de.hpc.ucar.edu 161:   0.000000000000000E+000  0.000000000000000E+000  0.000000000000000E+000
dec1322.hsn.de.hpc.ucar.edu 161:   0.000000000000000E+000  0.000000000000000E+000  0.000000000000000E+000
dec1322.hsn.de.hpc.ucar.edu 161:   0.000000000000000E+000  0.000000000000000E+000  0.000000000000000E+000
dec1322.hsn.de.hpc.ucar.edu 161:   0.000000000000000E+000  0.000000000000000E+000
dec1322.hsn.de.hpc.ucar.edu 161:  ENDRUN:
dec1322.hsn.de.hpc.ucar.edu 161:  ERROR in EDInitMod.F90 at line 922
```

Case D: `flandusepftdat = ''` and `use_fates_fixed_biogeog` true: Fails as B.

So I think we need to put up guardrails:
- Do not allow running (with `use_fates_luh` true?) with `flandusepftdat` unset.
- Do not allow running (with `use_fates_luh` true?) with `use_fates_fixed_biogeog` true.

But I don't know exactly how @ckoven thinks these should interact with each other: Maybe they should actually work?

Reply:
Thanks @samsrabin. I think all of these things (except maybe case D) should be permitted, but with different behavior. Probably a table is warranted to help keep track of what these different overlapping things mean. Maybe you, @glemieux and I need to discuss next week.

Reply:
@samsrabin I started to make a table of valid FATES land use change-enabled configurations, [here](https://docs.google.com/spreadsheets/d/1eE3sRMYxfocZKbT8uIQhXpjjtfM2feXPRSWXJNoo4jM/edit#gid=0). I think your case A should actually not be permitted if running with land use.